lib/bank_account/account.rb

```diff
@@ -0,0 +1,37 @@
+module BankAccount
+
+  class Account
+    attr_reader :id, :balance # attributes' default instance method
+
+    #class method
+    def initialize(id, initial_balance)
+      @id = id
+      @balance = initial_balance #local variable
+      # made initial balance equal to instance variable balance, which actually turns out to be the current balance
+
+      #this raises an error to prevent the initial balance from being lower than 0
+      unless @balance >= 0
+        raise ArgumentError.new("You can't start out with a negative balance. Add some monies")
+      end
+    end
+
+    def withdraw(amount)
+
+  # If the amount to withdraw is greater than the current balance, it throws and error and returns the original balance
+      unless @balance >= amount
```

Comment:
I tend to only use `unless` when it makes a ton of sense.  If I was telling someone what this function does (which you did in the comment above), I would use `if` instead.  So something like this instead:
```
if amount >= @balance
  puts "You can't overdraw. Your balance is still $#{@balance}."
```

Also, I'm noticing that this function returns @balance in the first part of the conditional, but the second part doesn't return @balance.  You don't need that line that says `return @balance`